packages/@atjson/source-html/src/converter/video-embeds.ts

```diff
@@ -0,0 +1,193 @@
+  );
+}
+
+function isBrightcoveEmbed(annotation: Annotation<any>) {
+  return (
+    is(annotation, Iframe) &&
+    annotation.attributes.src?.indexOf("https://players.brightcove.net") === 0
+  );
+}
+
+function isBrightcoveWrapper(annotation: Annotation<any>) {
+  return (
+    is(annotation, Div) && annotation.attributes.style?.match(/padding\-top/)
+  );
+}
+
+function isVimeoLink(annotation: Annotation<any>) {
+  return (
+    is(annotation, Anchor) &&
+    annotation.attributes.href?.indexOf("https://vimeo.com")
+  );
+}
+
+function getSize(annotation: Iframe, name: "width" | "height") {
+  let dimension = annotation.attributes[name];
+  if (dimension) {
+    return parseInt(dimension, 10);
+  }
+  return undefined;
+}
+
+export default function(doc: Document) {
+  doc
+    .where(isVimeoEmbed)
+    .as("video")
+    .outerJoin(
+      doc.where({ type: "-html-p" }).as("paragraph"),
+      function findPossibleVimeoCaptions(embed, paragraph) {
+        return (
+          paragraph.start === embed.end || paragraph.start === embed.end + 1
+        );
+      }
+    )
+    .outerJoin(doc.where(isVimeoLink).as("links"), function vimeoCaptionJoin(
+      { paragraph },
+      link
+    ) {
+      return covers(paragraph[0], link);
+    })
+    .update(function convertVimeoVideos({ video, paragraph }) {
+      let url = VideoURLs.identify(new URL(video.attributes.src));
+      assert(
+        url,
+        `The Vimeo embed ${video.attributes.src} was definitely defined in our queries, but was not identified.`
+      );
+
+      let caption: CaptionSource | undefined;
+      if (paragraph.length) {
+        caption = doc
+          .cut(paragraph[0].start, paragraph[0].end)
+          .convertTo(OffsetSource);
+      }
+
+      let width = getSize(video, "width");
+      let height = getSize(video, "height");
+      doc.replaceAnnotation(
+        video,
+        new VideoEmbed({
+          start: video.start,
+          end: video.end,
+          attributes: {
+            url,
+            width,
+            height,
+            aspectRatio:
+              width && height
+                ? getClosestAspectRatio(width, height)
+                : undefined,
+            caption
+          }
+        })
+      );
+    });
+
+  doc
+    .where(isBrightcoveEmbed)
+    .as("video")
+    .outerJoin(
+      doc.where(isBrightcoveWrapper).as("wrappers"),
+      function joinResponsiveDiv(video, div) {
+        return covers(div, video);
+      }
+    )
+    .outerJoin(
+      doc.where({ type: "-html-div" }).as("divsToToss"),
+      function joinWrappingDivs({ video }, div) {
+        return covers(div, video);
+      }
+    )
+    .update(function convertBrightcoveVideo({ video, wrappers, divsToToss }) {
+      let width: number | undefined;
+      let height: number | undefined;
+      if (wrappers[0]) {
+        let paddingTop = wrappers[0].attributes.style.match(
+          /padding\-top: ([.\d]+)/
```

Comment:
How confident are we that this will always be a `padding-top` with a fixed pixel value, and not, say, a `padding` or a relative value?

Reply:
I guess if it's not that, then it's safe to say we should ignore it which is what you're already doing, so this is good?